lib/thor/actions/empty_directory.rb

```diff
@@ -112,11 +112,18 @@ def invoke_with_conflict_check(&block)
         if exists?
           on_conflict_behavior(&block)
         else
-          say_status :create, :green
           yield unless pretend?
+          say_status :create, :green
         end
 
         destination
+      rescue SystemCallError => e
```

Comment:
It is better to rescue the specific exception here and avoid the `unless` instead of the parent class.